Make UsbBus more flexible and fix warnings throughout boards

bradleyharden opened the pull request:
Change the USB modules to accept the corrects pin in any mode. v2::Pins
do not need access to the Port to change mode, so also remove the Port
argument from corresponding BSP functions.

Fix warnings throughout the board modules, so that everything builds
cleanly.

@ryankurte, this implements the proposed solution in #404. But it ended up being a little more complicated than I originally thought. I'll expand on that in #404.

Edit: closes #404

bradleyharden commented:
Hold off on merging this. I just had a stroke of inspiration while brushing my teeth. I'll make more changes tomorrow

twitchyliquid64 commented:
HODL'ing.

> I just had a stroke of inspiration while brushing my teeth

:joy:

bradleyharden commented:
Better than having a stroke while brushing your teeth.

I added `usb` to the features of 6 more boards in `crates.json`, which should improve coverage for `build-all.py`.

I also simplified pin mode conversion. I'll update #404 with the simpler way to do it.
